taichi/transforms/offload.cpp

```diff
@@ -555,7 +556,8 @@ class FixCrossOffloadReferences : public BasicStmtVisitor {
 
     if (local_to_global_offset.find(op) == local_to_global_offset.end()) {
       TI_ASSERT_INFO(op->is<ConstStmt>() || op->is<PtrOffsetStmt>() ||
-                         op->is<GlobalTemporaryStmt>(),
+                         op->is<GlobalTemporaryStmt>() ||
+                         op->is<ArgLoadStmt>() || op->is<ExternalPtrStmt>(),
```

Comment:
nit: Should this also check for `ArgLoadStmt::is_ptr`?

Reply:
Yes that would be more strict and accurate. Fixed.